lib/ZumoOta/src/BootloaderCom.cpp

```diff
@@ -101,12 +102,84 @@ class IntProgCmdProvider: public CmdProvider {
     static const ResponseInfo m_responses[SEQ_LENGHT]; /**< Array of response information */
 };
 
-#if 0
+/**
+ * @class FwProgCmdProvider
+ * @brief Implementation of the command provider for firmware programming.
+ *
+ * This class provides commands for firmware programming, specifically tailored
+ * for the Zumo32U4 device.
+ */
 class FwProgCmdProvider : public CmdProvider {
     public:
-        FwProgCmdProvider(FileHandle * fs);
-}
-#endif
+        /**
+         * @brief Constructor for the FwProgCmdProvider class.
+         * @param fs A pointer to the FileHandler object for handling file operations.
+         */
+        FwProgCmdProvider(FileHandler * fs) :
+        m_index(0),
+        m_fileSystem(fs)
+    {}
+
+    /**
+     * @brief Resets the command provider to its initial state.
+     *
+     * This method resets the internal index that tracks the current command
+     * in the command sequence memory.
+     */
+    virtual void reset() { m_index = 0; }
+
+    /**
+     * @brief Retrieves the next command and the next response.
+     *
+     * This method returns the next command and its associated response
+     * from the command sequence memory.
+     *
+     * @param cmd Reference to a pointer to the next command.
+     * @param rsp Reference to a pointer to the next response.
+     * @return True if there is a next command and next response, false otherwise.
+     */
+    virtual bool next(const CommandInfo *& cmd, const ResponseInfo *& rsp)
+    {
+        static const uint32_t CHUNK_SIZE = 128U;
+        /*Determine the current chunk start address.*/
+        uint32_t chunkStartAddress = m_index * CHUNK_SIZE;
+
+        /*Read 128 bytes from the file system.*/
+        char buffer[CHUNK_SIZE];

```

Comment:
That cannot work. Remember the cmd processing is in the background, so you cannot put data on the stack for background operations.